src/components/ActionButton/index.test.tsx

```diff
@@ -0,0 +1,54 @@
+import ActionButton from './';
+import { render, fireEvent} from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+
+describe('ActionButton Component', () => {
+  const actionBtn = (
+      <ActionButton
+        onClick={() => true}
+        messageDuration={300}
+        clickContent={<span>Some action occurred.</span>}
+        hoverContent={<span>Hovering message.</span>}
+        infotipProps={{
+          arrow: false,
+          direction: 'right'
+        }}
+        data-testid="action-button"
+      >
+        TEST
+      </ActionButton>
+  );
+
+  it('renders without errors', async () => {
+    const container = render(actionBtn);
+    const result = await container.findByTestId("action-button");
+    expect(result).toBeInTheDocument();
+  });
+  it('shows hover tooltip', async () => {
+    const container = render(actionBtn);
+    fireEvent.mouseOver(container.getByText("TEST"));
+    await new Promise((r) => setTimeout(r, 50));
```

Comment:
react-testing-library has some nice APIs for these situations: https://testing-library.com/docs/dom-testing-library/api-async/ you shouldn't need a timeout promise.